ui/ToggleGroup.tsx

```diff
@@ -0,0 +1,77 @@
+import { Dispatch, ReactNode, SetStateAction } from "react";
+import { RadioGroup, RadioOptionProps } from "@headlessui/react";
+import { cva } from "class-variance-authority";
+
+export const toggleOptionStyles = cva(
+  [
+    "flex items-center py-2 font-medium leading-5 rounded-12",
+    "outline-none cursor-pointer",
+  ],
+  {
+    variants: {
+      active: {
+        true: ["bg-surface-surface-0 shadow-2"],
+        false: ["hover:bg-surface-surface-1 hover:shadow-1"],
+      },
+      size: {
+        lg: ["py-3 px-4 text-md"],
+        md: ["py-2 px-3 text-base"],
+        sm: ["py-2 px-2 text-sm"],
+        xs: ["py-1 px-2 text-2xs"],
+      },
+    },
+    defaultVariants: {
+      size: "sm",
+    },
+  }
+);
+
+export type ToogleOptionSizeProp = "xs" | "sm" | "md" | "lg";
+
+interface ToggleGroupProp {
+  value: string;
+  onChange: Dispatch<SetStateAction<string>>;
+  children: React.ReactNode;
+}
+
+export const ToggleGroup = ({
+  value,
+  onChange,
+  children,
+  ...props
+}: ToggleGroupProp) => {
```

Comment:
```suggestion
export const ToggleGroup = (
  props: RadioGroupProps<React.ElementType, string>
) => {
```
We can use this types here, and we just need to spread props to `RadioGroup`